streaming/dataset.py

```diff
@@ -479,6 +480,22 @@ def __init__(
         self.stream_per_shard = np.array(stream_per_shard, np.int64)
         self.num_shards = len(self.shards)
 
+        # Maybe check all schemas match.
+        if not allow_schema_mismatch:
+            sigs = [shard.get_logical_type_signature() for shard in self.shards]
+            sig2count = Counter(sigs)
+            if len(sig2count) != 1:
```

Comment:
Are mixed logical types of the same kind (like int32 + int64) not allowed?

Also, the checking is

Reply:
Not allowed at this time for MVP. Maybe in the future, we would go with the wider type for all shards? But that would be kind of magical and might interact with custom getitem work?